src/promptflow/promptflow/_sdk/entities/_flow.py

```diff
@@ -166,9 +167,11 @@ def load(
         if flow_path.exists():
             # TODO: for file, we should read the yaml to get code and set path to source_path
             # read flow file to get hash
+            yaml = YAML()
+            yaml.preserve_quotes = True
             with open(flow_path, "r", encoding=DEFAULT_ENCODING) as f:
                 flow_content = f.read()
-                flow_dag = yaml.safe_load(flow_content)
+                flow_dag = yaml.load(flow_content)
```

Comment:
I see this pattern a lot of times. could there be a safe_load method in the utils?

Reply:
This part differs from our `load_yaml` util function which reads a local path or a readable stream object, so I added a new util function `load_yaml_string` to load yaml string.